### PR TITLE
support Authentication

### PR DESCRIPTION
**Pull request by FredLL-Avaiga**
add Login control
connected to taipy-enterprise #313 & #238

```
...
message = None

def on_login(state: State, id: str, payload):
    try:
        state.message = ""
        login(state, payload.get("args")[0], payload.get("args")[1])

        with Authorize(get_credentials(state)):
            ... do authenticated Core things

    except Exception as e:
        state.message = f"Error: {e}"

    navigate(state, to=payload.get("args")[2], force=True)


def on_logout(state: State, id: str, payload):
    logout(state)
    navigate(state, force=True)


login_md = """
<|login|on_action=on_login|message={message}|>
"""

pages = {
    "/": AuthorizedPage(
        IsAuthenticated("<|toggle|theme|><br/><|Log-out|button|on_action=on_logout|><br/><|navbar|>", login_md)
    ),
    "Scenario": AuthorizedPage(IsAuthenticated(scenario_md, login_md)),
    "ScenarioHtml": AuthorizedPage(IsAuthenticated(Html(scenario_html), login_md)),
    "Datanode": AuthorizedPage(IsAuthenticated(datanode_selector, login_md)),
    # "Comparison": comparison_md,
}
...
```

### CODE REVIEW

**Review by dinhlongviolin1 — DISMISSED** [review]
Tx for adding all the tests

File: taipy/gui/viselements.json [hunk 1]
```diff
@@ -761,6 +761,31 @@
...
+                "name": "on_action",
+                "type": "Callback",
+                "default_value": "on_login",
+                "doc": "The name of the function that is triggered when the dialog's button is pressed.<br/><br/>All the parameters of that function are optional:\n<ul>\n<li>state (<code>State^</code>): the state instance.</li>\n<li>id (str): the identifier of the button.</li>\n<li>payload (dict): the details on this callback's invocation.<br/>\nThis dictionary has the following keys:\n<ul>\n<li>action: the name of the action that triggered this callback.</li>\n<li>args: List(user, password, current page).</li>\n</ul>\n</li>\n</ul>",
+                "signature": [["state", "State"], ["id", "str"], ["payload", "dict"]]
+              },
+              {
+                "name": "message",
+                "type": "dynamic(str)",
+                "doc": "A message that is shown in the dialog."
```
Comment (FabienLelaquais) [inline taipy/gui/viselements.json]:
```suggestion
                "doc": "The message shown in the dialog."
```

File: taipy/gui/viselements.json [hunk 2]
```diff
@@ -761,6 +761,31 @@
...
       }
     ],
+    ["login", {
+        "inherits": ["shared"],
+        "properties": [
+            {
+                "name": "title",
+                "default_property": true,
+                "type": "str",
+                "default_value": "Log-in",
```
Comment (FabienLelaquais) [inline taipy/gui/viselements.json]:
```suggestion
                "default_value": "Log in",
```

File: taipy/gui/viselements.json [hunk 3]
```diff
@@ -761,6 +761,31 @@
...
+                "doc": "The title of the login dialog."
+            },
+            {
+                "name": "on_action",
+                "type": "Callback",
+                "default_value": "on_login",
+                "doc": "The name of the function that is triggered when the dialog's button is pressed.<br/><br/>All the parameters of that function are optional:\n<ul>\n<li>state (<code>State^</code>): the state instance.</li>\n<li>id (str): the identifier of the button.</li>\n<li>payload (dict): the details on this callback's invocation.<br/>\nThis dictionary has the following keys:\n<ul>\n<li>action: the name of the action that triggered this callback.</li>\n<li>args: List(user, password, current page).</li>\n</ul>\n</li>\n</ul>",
```
Comment (FabienLelaquais) [inline taipy/gui/viselements.json]: Is the id the id of the button, or the login control? Wouldn't it make more sense?

Comment (FabienLelaquais) [reply taipy/gui/viselements.json]:
```to be unformatted...
                <li>args: a tuple with three elements:
                    <ul>
                      <li>The first element is the user name</li>
                      <li>The second element is the password</li>
                      <li>The third element is the current page name</li>
                   </ul>
                 </li>
```

**Review by FabienLelaquais — DISMISSED** [review]
Looking pretty simple and nice!

**Review by dinhlongviolin1 — APPROVED** [review]

**Pull request merged**